src/details/ArborX_Polytope.hpp

```diff
@@ -0,0 +1,270 @@
+/****************************************************************************
+ * Copyright (c) 2012-2020 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef ARBORX_POLYTOPE_HPP
+#define ARBORX_POLYTOPE_HPP
+
+#include <ArborX_DetailsKokkosExt.hpp> // ArithmeticTraits
+#include <ArborX_DetailsTags.hpp>
+#include <ArborX_Point.hpp>
+
+#include <Kokkos_Macros.hpp>
+
+namespace ArborX
+{
+
+namespace Details
+{
+struct Direction
+{
+  float _data[3];
+};
+KOKKOS_FUNCTION float project(Point const &p, Direction const &d)
+{
+  float r = 0.;
+  for (int i = 0; i < 3; ++i)
+  {
+    r += p[i] * d._data[i];
+  }
+  return r;
+}
+} // namespace Details
+
+template <typename T>
+struct Polytope : public T
+{
+  static constexpr int n_directions = T::directions.size();
+  Kokkos::Array<float, n_directions> _min_values;
+  Kokkos::Array<float, n_directions> _max_values;
+  KOKKOS_FUNCTION Polytope()
+  {
+    for (int i = 0; i < n_directions; ++i)
+    {
+      _min_values[i] = KokkosExt::ArithmeticTraits::max<float>::value;
+      _max_values[i] = -KokkosExt::ArithmeticTraits::max<float>::value;
+    }
+  }
+  KOKKOS_FUNCTION Polytope &operator+=(Point const &p)
+  {
+    using KokkosExt::max;
+    using KokkosExt::min;
+    for (int i = 0; i < n_directions; ++i)
+    {
+      auto const proj_i = Details::project(p, this->directions[i]);
+      _min_values[i] = min(_min_values[i], proj_i);
+      _max_values[i] = max(_max_values[i], proj_i);
+    }
+    return *this;
+  }
+  KOKKOS_FUNCTION Polytope &operator+=(Polytope const &other)
+  {
+    using KokkosExt::max;
+    using KokkosExt::min;
+    for (int i = 0; i < n_directions; ++i)
+    {
+      _min_values[i] = min(_min_values[i], other._min_values[i]);
+      _max_values[i] = max(_max_values[i], other._max_values[i]);
+    }
+    return *this;
+  }
+  KOKKOS_FUNCTION bool intersects(Point const &p) const
+  {
+    for (int i = 0; i < n_directions; ++i)
+    {
+      auto const proj_i = Details::project(p, this->directions[i]);
+      if (proj_i < _min_values[i] || proj_i > _max_values[i])
+      {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
Note that this is an approximate intersections. In certain situations, such as if there axes that are missing the orthogonal direction, the polytopes may still not intersect if if this implementation returns true.

Reply:
I take it back.